### PR TITLE
Finished workflows for testing and linting on push/pull request to main.

### PR DESCRIPTION
**Pull request by CalebMathers**
Fixes issue #26 

Basic workflows written to test and lint the transfer pipeline, main pipeline, and visualisation directories.

### CODE REVIEW

**Review by mehrdadhalali — CHANGES_REQUESTED** [review]
Rewrite the entire thing in assembly for efficiency.

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  test:
+    name:    "Run tests"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Install python"
+        uses: actions/setup-python@v5.1.1
+        with:
+          python-version: '3.12'
+      - name: "Get repo code"
+        uses: actions/checkout@v4
```
Comment (megansam5) [inline .github/workflows/main.yaml:20]: nice remembering to use v4

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      - name: "Test data-transfer-pipeline"
+        run:  |
+              cd data-transfer-pipeline/
+              echo "SCHEMA_NAME=${{ secrets.SCHEMA_NAME }}" > .env
+              pip install -r requirements.txt
+              pytest
+              echo 'Data transfer pipeline tests finished.'
+              cd ..
+      - name: "Test pipeline"
```
Comment (megansam5) [inline .github/workflows/main.yaml:29]: i like the splitting into files

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          python-version: '3.12'
+      - name: "Get repo code"
+        uses: actions/checkout@v4
+      - name: "Install pylint"
+        run:  pip install pylint
+      - name: "Lint data-transfer-pipeline"
+        run:  |
+              cd data-transfer-pipeline/
+              pip install -r requirements.txt
+              pylint *.py --fail-under=8
```
Comment (megansam5) [inline .github/workflows/main.yaml:59]: good failing level

**Pull request merged**